hook-janitor/src/webhooks.rs

```diff
@@ -315,33 +346,51 @@ impl WebhookCleaner {
 
         let mut tx = self.start_serializable_txn().await?;
 
-        let completed_agg_row_count = {
-            let completed_rows = self.get_completed_rows(&mut tx).await?;
-            let row_count = completed_rows.len();
+        let (completed_row_count, completed_agg_row_count) = {
+            let completed_row_count = self.get_row_count_for_status(&mut tx, "completed").await?;
+            let completed_agg_rows = self.get_completed_agg_rows(&mut tx).await?;
+            let agg_row_count = completed_agg_rows.len() as u64;
             let completed_app_metrics: Vec<AppMetric> =
-                completed_rows.into_iter().map(Into::into).collect();
+                completed_agg_rows.into_iter().map(Into::into).collect();
             self.send_metrics_to_kafka(completed_app_metrics).await?;
-            row_count
+            (completed_row_count, agg_row_count)
         };
 
-        let failed_agg_row_count = {
-            let failed_rows = self.get_failed_rows(&mut tx).await?;
-            let row_count = failed_rows.len();
+        let (failed_row_count, failed_agg_row_count) = {
+            let failed_row_count = self.get_row_count_for_status(&mut tx, "failed").await?;
+            let failed_agg_rows = self.get_failed_agg_rows(&mut tx).await?;
+            let agg_row_count = failed_agg_rows.len() as u64;
             let failed_app_metrics: Vec<AppMetric> =
-                failed_rows.into_iter().map(Into::into).collect();
+                failed_agg_rows.into_iter().map(Into::into).collect();
             self.send_metrics_to_kafka(failed_app_metrics).await?;
-            row_count
+            (failed_row_count, agg_row_count)
         };
 
-        let mut rows_processed = 0;
+        let mut rows_deleted = 0;
         if completed_agg_row_count + failed_agg_row_count != 0 {
-            rows_processed = self.delete_observed_rows(&mut tx).await?;
+            rows_deleted = self.delete_observed_rows(&mut tx).await?;
+
+            if rows_deleted != completed_row_count + failed_row_count {
+                // This should never happen, but if it does, we want to know about it (and abort the
+                // txn).
```

Comment:
I'm somewhat afraid that we are not aborting the transaction here. Since we take a `&mut` instead of taking ownership of `tx` the caller could still commit the transaction even on error. Could we take ownership of `tx` and call `rollback` here? Similar to `commit_txn` below.

Reply:
I agree that this should never happen, but in the event it does, I would like to ensure we clean up properly :sparkles:

Reply:
Good idea, I shouldn't depend on `Drop` for something so dangerous.

Reply:
Thank you!